assessment/collectors/project_analyzer/file_dependency_analyzer.go

```diff
@@ -56,6 +57,35 @@ func packagesLoadLogger(format string, args ...interface{}) {
 	logger.Log.Debug(fmt.Sprintf(format, args...))
 }
 
+func (b *BaseAnalyzer) RemoveCycle(fileDependenciesMapWithCycle map[string]map[string]struct{}) map[string]map[string]struct{} {
```

Comment:
The remove cycles should have been there in golang as well, how is this different ?

Reply:
I have scrapped the required code for removing cycle from GoDependencyAnalyzer and moved it to BaseAnalyzer.

Hence, this can be used by all the dependency analyzers.